domain/controllerconfig/service/service.go

```diff
@@ -196,6 +211,13 @@ func contains(s []string, e string) bool {
 	return false
 }
 
+func hashCompleteS3Config(config map[string]string) bool {
```

Comment:
Couldn't you coerce and use the one from the controller config package? Function is misspelled in any case.